src/FMBot.Bot/Builders/DiscogsBuilder.cs

```diff
@@ -223,4 +225,124 @@ public async Task<ResponseModel> DiscogsLoginAsync(ContextModel context)
         response.ResponseType = ResponseType.Paginator;
         return response;
     }
+
+    public async Task<ResponseModel> DiscogsTopArtistsAsync(
+        ContextModel context,
+        TopListSettings topListSettings,
+        TimeSettingsModel timeSettings,
+        UserSettingsModel userSettings)
+    {
+        var response = new ResponseModel
+        {
+            ResponseType = ResponseType.Paginator,
+        };
+
+        var user = await this._userService.GetUserWithDiscogs(userSettings.DiscordUserId);
+
+        if (user.UserDiscogs == null)
+        {
+            if (!userSettings.DifferentUser)
+            {
+                response.Embed.WithDescription("To use the top artists commands with Discogs you have to connect a Discogs account.\n\n" +
+                                               $"Use the `{context.Prefix}discogs` command to get started.");
+            }
+            else
+            {
+                response.Embed.WithDescription("The user you're trying to look up has not setup their Discogs account yet.");
+            }
+
+            response.CommandResponse = CommandResponse.UsernameNotSet;
+            return response;
+        }
+
+        if (user.UserDiscogs.ReleasesLastUpdated == null ||
+            user.UserDiscogs.ReleasesLastUpdated <= DateTime.UtcNow.AddHours(-1))
+        {
+            user.UserDiscogs = await this._discogsService.StoreUserReleases(user);
+            user.UserDiscogs = await this._discogsService.UpdateCollectionValue(user.UserId);
+        }
+
+        var pages = new List<PageBuilder>();
+
+        string userTitle;
+        if (!userSettings.DifferentUser)
+        {
+            if (!context.SlashCommand)
+            {
+                response.EmbedAuthor.WithIconUrl(context.DiscordUser.GetAvatarUrl());
+            }
+            userTitle = await this._userService.GetUserTitleAsync(context.DiscordGuild, context.DiscordUser);
+        }
+        else
+        {
+            userTitle =
+                $"{user.UserDiscogs.Username}, requested by {await this._userService.GetUserTitleAsync(context.DiscordGuild, context.DiscordUser)}";
+        }
+        var userUrl =
+            $"{Constants.DiscogsUserURL}{user.UserDiscogs.Username}/collection";
+
+        response.EmbedAuthor.WithName($"Top {timeSettings.Description.ToLower()} Discogs artists for {userTitle}");
+        response.EmbedAuthor.WithUrl(userUrl);
+
+        var topArtists = new Dictionary<String, TopDiscogsArtist>();
+
+        foreach (UserDiscogsReleases item in user.DiscogsReleases)
+        {
+            if (item.DateAdded < timeSettings.StartDateTime || item.DateAdded >= timeSettings.EndDateTime) {
+                continue;
+            }
+            TopDiscogsArtist value = null;
+            if (!topArtists.TryGetValue(item.Release.Artist, out value))
+            {
+                value.ArtistName = item.Release.Artist;
```

Comment:
it accesses `value` here when the `TryGetValue` fails, resulting in an exception since value is null

Reply:
i'm a little bit confused with this part (rather, i don't really get TryGetValue although it's touted as one of the quickest ways to check for an element in a dictionary). i thought `out value` would give me a `default(TopDiscogsArtist)`, but maybe that doesn't work if it hasn't been implemented :thinking: